packages/react-shadow/src/root.tsx

```diff
@@ -24,10 +32,33 @@ const ReactComponentsWrapper: React.FC<{
         return;
       }
 
-      root.adoptedStyleSheets = [
-        ...root.adoptedStyleSheets,
-        ...renderer.adoptedStyleSheets,
-      ];
+      const insertionIndex = root.adoptedStyleSheets.findIndex(
+        (styleSheet) =>
+          (styleSheet as ExtendedCSSStyleSheet).metadata?.[
+            REACT_SHADOW_INSERTION_BUCKET_NAME
+          ] === true
+      );
+
+      if (insertionIndex !== -1) {
+        if (SUPPORTS_MODIFYING_ADOPTED_STYLESHEETS) {
```

Comment:
mutable stylesheets introduce and interesting problem - if users prepend/append stylesheets this effect will never run because the reference to the adopted stylesheet never changes.

However currently there is a bunch of conditions at the start of the effect that essentially stop any updates to adopted stylesheets anyway...

Reply:
We should probably work out how this bit works because it also addresses a bug in Safari: https://github.com/microsoft/fast/pull/6703

That said, maybe it's best to drop it from this PR and follow up later?

Reply:
I've removed this for later consideration.